src/platforms/posix/drivers/df_lsm9ds1_wrapper/df_lsm9ds1_wrapper.cpp

```diff
@@ -671,13 +665,13 @@ int DfLsm9ds1Wrapper::_publish(struct imu_sensor_data &data)
 	accel_report.z = accel_val_filt(2);
 
 	if (_mag_enabled) {
-
+		// the sensor coordinate system of mag is different. MAG_X should be -MAG_X
+		data.mag_ga_x = - data.mag_ga_x;
```

Comment:
@julianoes thanks. There is a picture of sensor coordinate system in Page10 [LSM9DS1 datasheet](http://www.st.com/content/ccc/resource/technical/document/datasheet/1e/3f/2a/d6/25/eb/48/46/DM00103319.pdf/files/DM00103319.pdf/jcr:content/translations/en.DM00103319.pdf). Is it correct?  Here is [PR](https://github.com/PX4/DriverFramework/pull/160) for this.